### PR TITLE
[Merged by Bors] - feat: port Data.Rat.NNRat

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Rat/NNRat.lean [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+
+
+open Function
+
+open BigOperators
+
+/-- Nonnegative rational numbers. -/
+def NNRat := { q : ℚ // 0 ≤ q } deriving
+  CanonicallyOrderedCommSemiring, CanonicallyLinearOrderedSemifield, LinearOrderedCommGroupWithZero,
+  Sub, Inhabited  -- Porting note: Removed `OrderedSub, DenselyOrdered, Archimedean`
```
Comment (jcommelin) [inline Mathlib/Data/Rat/NNRat.lean:38]: Why did you remove these? Are they implied by the others?

Comment (int-y1) [reply Mathlib/Data/Rat/NNRat.lean]: They were causing errors. I used `instance` to try to add them back.

File: Mathlib/Data/Rat/NNRat.lean [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+
+open BigOperators
+
+/-- Nonnegative rational numbers. -/
+def NNRat := { q : ℚ // 0 ≤ q } deriving
+  CanonicallyOrderedCommSemiring, CanonicallyLinearOrderedSemifield, LinearOrderedCommGroupWithZero,
+  Sub, Inhabited
+#align nnrat NNRat
+
+-- Porting note: Added these instances to get `OrderedSub, DenselyOrdered, Archimedean`
```
Comment (jcommelin) [inline Mathlib/Data/Rat/NNRat.lean:41]:
```suggestion
-- Porting note: Added these instances to get `OrderedSub, DenselyOrdered, Archimedean`
-- instead of `deriving` them
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge
